tests/components/element-editor/basic-details/basic-details.spec.tsx

```diff
@@ -0,0 +1,169 @@
+            renderComponent({
+                builderContext: {
+                    focusedElement: MOCK_FOCUSED_ELEMENT,
+                    elements: MOCK_ELEMENTS,
+                },
+            });
+            const idInput = await getIdField();
+            fireEvent.focus(idInput);
+            fireEvent.blur(idInput);
+            const idErrorMessage = await getIdErrorMessage();
+            expect(idErrorMessage).toHaveTextContent("ID is required");
+        });
+
+        it("should render an error message for the ID field when input is invalid", async () => {
+            renderComponent({
+                builderContext: {
+                    focusedElement: MOCK_FOCUSED_ELEMENT,
+                    elements: MOCK_ELEMENTS,
+                },
+            });
+            const idInput = await getIdField();
+            fireEvent.focus(idInput);
+            fireEvent.change(idInput, { target: { value: "camel_Case" } });
+            fireEvent.blur(idInput);
+            const idErrorMessage = await getIdErrorMessage();
+            expect(idErrorMessage).toHaveTextContent("ID must be camelCase");
+        });
+
+        it("should render an error message for the label field", async () => {
+            renderComponent({
+                builderContext: {
+                    focusedElement: MOCK_FOCUSED_ELEMENT,
+                    elements: MOCK_ELEMENTS,
+                },
+            });
+            const labelInput = await getLabelField();
+            fireEvent.focus(labelInput);
+            fireEvent.change(labelInput, { target: { value: "" } });
+            fireEvent.blur(labelInput);
+            console.log("check this:", document.body.innerHTML);
+            const labelErrorMessage = await getLabelErrorMessage();
+            expect(labelErrorMessage).toHaveTextContent("Label is required");
+        });
+    });
+});
+
+// =============================================================================
+// HELPER FUNCTIONS
+// =============================================================================
+
+const MyTestComponent = () => {
+    const methods = useForm({
+        mode: "onTouched",
+        resolver: yupResolver(SchemaHelper.buildSchema(EElementType.EMAIL)),
+    });
+    return (
+        <FormProvider {...methods}>
+            <BasicDetails />
+        </FormProvider>
+    );
+};
+
+const renderComponent = (overrideOptions?: TestHelper.RenderOptions) => {
+    return render(
+        TestHelper.withProviders(overrideOptions, <MyTestComponent />)
+    );
+};
+
+const getIdField = async () => {
+    return screen.findByTestId("id-field");
+};
+
+const getLabelField = async () => {
+    return screen.findByTestId("label-field");
+};
+
+const getIdErrorMessage = async () => {
+    return screen.findByTestId("form-field-error-message");
+};
+
+const getLabelErrorMessage = async () => {
+    return screen.findByTestId("form-textarea-error-message");
+};
```

Comment:
although `findByTestId` works, it is not ideal based on react-testing-library's [guiding principles](https://testing-library.com/docs/queries/about/#priority), may I trouble you explore following the priority as described in their docs?